transform/run_pipelines.sh

```diff
@@ -0,0 +1,59 @@
+#!/bin/bash
+#  Copyright (c) University College London Hospitals NHS Foundation Trust
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -o errexit
+set -o pipefail
+set -o nounset
+# Uncomment this line to see each command for debugging (careful: this will show secrets!)
+# set -o xtrace
+
+SCRIPT_DIR=$( cd -- "$( dirname -- "${BASH_SOURCE[0]}" )" &> /dev/null && pwd )
+FACTORY_NAME=$(cd "${SCRIPT_DIR}"/../infrastructure/transform && terragrunt output adf_name | tr -d '"')
+RESOURCE_GROUP_NAME=$(cd "${SCRIPT_DIR}"/../infrastructure/core && terragrunt output core_rg_name | tr -d '"')
+export FACTORY_NAME
```

Comment:
are we exporting these because of the `parallel` call?
can't we pass them as params to `run_pipeline_and_wait` rather than set them globally?

Reply:
`parallel` starts subshells so these have to be exported

Reply:
Yeah, but there's no param passing?

Reply:
It's coming from the pipe:

```bash
if ! echo "${pipelines}" | parallel run_pipeline_and_wait; then
    echo "One or more pipeline runs have failed - please check the logs"
    exit 1
fi
```

This code will just run `run_pipeline_and_wait` once per each pipeline

Reply:
Ah yes. OK. 

So, related to my other question, couldn't the 2 `export` params be passed in via the same mechanism?

Reply:
It probably can but I can't quite work out how. I think `parallel` works best when you give it an array-like thing and it runs a process for each instance of that array.  I was using this to write it: https://www.gnu.org/software/parallel/parallel_examples.html#example-rewriting-a-for-loop-and-a-while-read-loop

Unless you think there's a big concern with this `export` statement, I would be reluctant to change it. If I make big changes to this script, I would probably just rewrite it in Python as James suggested above.

Reply:
I think you'd just add the values to your echo (where you pass the pipelines value), but I agree...it's not a big deal